packages/Search/test/Search_UnitTestHelpers.hpp

```diff
@@ -0,0 +1,220 @@
+    TEST_COMPARE_ARRAYS( offset_host, offset_ref );
+    TEST_COMPARE_ARRAYS( ranks_host, ranks_ref );
+    TEST_COMPARE_FLOATING_ARRAYS( distances_host, distances_ref, 1e-14 );
+}
+
+template <typename DeviceType>
+DataTransferKit::BVH<DeviceType>
+makeBvh( std::vector<DataTransferKit::Box> const &b )
+{
+    int const n = b.size();
+    Kokkos::View<DataTransferKit::Box *, DeviceType> boxes( "boxes", n );
+    auto boxes_host = Kokkos::create_mirror_view( boxes );
+    for ( int i = 0; i < n; ++i )
+        boxes_host( i ) = b[i];
+    Kokkos::deep_copy( boxes, boxes_host );
+    return DataTransferKit::BVH<DeviceType>( boxes );
+}
+
+template <typename DeviceType>
+DataTransferKit::DistributedSearchTree<DeviceType>
+makeDistributedSearchTree( Teuchos::RCP<const Teuchos::Comm<int>> const &comm,
+                           std::vector<DataTransferKit::Box> const &b )
+{
+    int const n = b.size();
+    Kokkos::View<DataTransferKit::Box *, DeviceType> boxes( "boxes", n );
+    auto boxes_host = Kokkos::create_mirror_view( boxes );
+    for ( int i = 0; i < n; ++i )
+        boxes_host( i ) = b[i];
+    Kokkos::deep_copy( boxes, boxes_host );
+    return DataTransferKit::DistributedSearchTree<DeviceType>( comm, boxes );
+}
+
+// Hopefully we can git rid of this once the operator== has been implemented
+// for boxes.
+void testBoxEquality( DataTransferKit::Box const &l,
+                      DataTransferKit::Box const &r, bool &success,
+                      Teuchos::FancyOStream &out )
+{
+    TEST_EQUALITY( l[0], r[0] );
+    TEST_EQUALITY( l[1], r[1] );
+    TEST_EQUALITY( l[2], r[2] );
+    TEST_EQUALITY( l[3], r[3] );
+    TEST_EQUALITY( l[4], r[4] );
+    TEST_EQUALITY( l[5], r[5] );
+}
+
+template <typename DeviceType>
+Kokkos::View<DataTransferKit::Details::Overlap *, DeviceType>
+makeOverlapQueries( std::vector<DataTransferKit::Box> const &boxes )
+{
+    int const n = boxes.size();
+    Kokkos::View<DataTransferKit::Details::Overlap *, DeviceType> queries(
+        "overlap_queries", n );
+    auto queries_host = Kokkos::create_mirror_view( queries );
+    for ( int i = 0; i < n; ++i )
+        queries_host( i ) = DataTransferKit::Details::overlap( boxes[i] );
+    Kokkos::deep_copy( queries, queries_host );
+    return queries;
+}
+
+template <typename DeviceType>
+Kokkos::View<DataTransferKit::Details::Nearest *, DeviceType>
+makeNearestQueries(
+    std::vector<std::pair<DataTransferKit::Point, int>> const &points )
+{
+    // NOTE: `points` is not a very descriptive name here. It stores both the
+    // actual point and the number k of neighbors to query for.
+    int const n = points.size();
+    Kokkos::View<DataTransferKit::Details::Nearest *, DeviceType> queries(
+        "nearest_queries", n );
+    auto queries_host = Kokkos::create_mirror_view( queries );
+    for ( int i = 0; i < n; ++i )
+        queries_host( i ) = DataTransferKit::Details::nearest(
+            points[i].first, points[i].second );
```

Comment:
What happens if `points[i].second` is -1?

Reply:
The search result will have `offset(i) == offset(i+1)`, i.e. it will tell you it did not find anything.
The construction of the nearest predicate will not complain since it only stores `k` and the query will not traverse the tree and return (See [here](https://github.com/ORNL-CEES/DataTransferKit/blob/master/packages/Search/src/details/DTK_DetailsTreeTraversal.hpp#L173-L199)